### DISCUSSION TITLE
How to build AI image recognition and matching platform?

### DISCUSSION THREAD

**Discussion by diligent423** [community/community#148188] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I'm building platform that managing image recognition and video analytics on GCP.
The data is captured from drones and want to implement this function in real-tiem.
I need help and I want to know if anyone has open-source.

**Comment by 0x4k-helios** [comment]
That sounds like a great project! For real-time image recognition and video analytics, Google Cloud has tools that can help. You can use its services to stream data from your drones, process it in real-time, and apply machine learning models to analyze the images and videos. Tools like Pub/Sub and Dataflow are great for handling live data streams, and Vertex AI can help with building and deploying machine learning models.

For open-source solutions, you might want to check GitHub for pre-trained models or libraries for image recognition and video processing, like TensorFlow or OpenCV. These can save you a lot of time in development.

If you have specific challenges, feel free to share them, and I can help with more targeted advice!

**Reply by diligent423** [reply]
Hi, @0x4k-helios
Thanks you for your help.
I also want to know if Google Cloud Platform Vertex AI supports real-time video analytic, according my research, the Vertex Ai does not support.

**Reply by 0x4k-helios** [reply, marked as answer]
Well, actually Google Cloud Platform Vertex Ai does not support real-time video directly, so you have to use Google Cloud Pub/Sub or Cloud Storage, Dataflow for real-time analytic.

**Comment by divyamlathiya** [comment]
You can explore these open-source projects for real-time image recognition and video analytics:

**1. TensorFlow + TensorFlow Lite** – for real-time image classification and inference.
**2. OpenCV** – for video processing and real-time computer vision tasks.
**3. DeepStream SDK by NVIDIA** – for real-time video analytics (works on GCP with NVIDIA GPUs).
**4. Apache Kafka + Apache Flink** – for real-time data processing and streaming.
**5. YOLO (You Only Look Once)** – for fast, real-time object detection.
These can be integrated with GCP services like **Google Cloud Storage**, **AI Platform**, and **Compute Engine** for scalable solutions.
